File: solutions/php/high-scores/1/HighScores.php
<?php

declare(strict_types=1);

class HighScores
{
    public array $scores;
    public int $latest;
    public int $personalBest;
    public array $personalTopThree;

    public function __construct(array $scores)
    {
        $this->scores = $scores;
        $this->latest = $scores[array_key_last($scores)];

        $sorted = $scores;
        rsort($sorted);

        $this->personalBest = $sorted[0];
        $this->personalTopThree = array_slice($sorted, 0, 3, false);
    }
}

File: solutions/python/high-scores/1/high_scores.py
from heapq import nlargest


class HighScores:
    def __init__(self, scores: list[int]) -> None:
        self._scores = scores[:]

    @property
    def scores(self) -> list[int]:
        return self._scores

    def latest(self) -> int | None:
        return self._scores[-1] if self._scores else None

    def personal_best(self) -> int | None:
        return max(self._scores) if self._scores else None

    def personal_top_three(self) -> list[int]:
        return nlargest(3, self._scores)
